app/Controllers/extensionController.php

```diff
@@ -124,26 +126,36 @@ public function enableAction() {
 				Minz_Request::bad(_t('feedback.extensions.already_enabled', $ext_name), $url_redirect);
 			}
 
+			$type = $ext->getType();
+			if ($type !== 'user' && !FreshRSS_Auth::hasAccess('admin')) {
+				Minz_Request::bad(_t('feedback.extensions.no_access', $ext_name), $url_redirect);
+				return;
+			}
+
 			$conf = null;
-			if ($ext->getType() === 'system' && FreshRSS_Auth::hasAccess('admin')) {
+			if ($type === 'system') {
 				$conf = FreshRSS_Context::$system_conf;
-			} elseif ($ext->getType() === 'user') {
+			} elseif ($type === 'user') {
 				$conf = FreshRSS_Context::$user_conf;
-			} else {
-				Minz_Request::bad(_t('feedback.extensions.no_access', $ext_name), $url_redirect);
 			}
 
 			$res = $ext->install();
 
 			if ($res === true) {
 				$ext_list = $conf->extensions_enabled;
+				$ext_list = array_filter($ext_list, function($key) use($type) {
+					// Remove from list the extensions that have disappeared or changed type
+					$extension = Minz_ExtensionManager::findExtension($key);
+					return $extension !== null && $extension->getType() === $type;
+				}, ARRAY_FILTER_USE_KEY);
```

Comment:
We had a case of when extensions were deleted, they would stay enabled for ever in the configuration.

Reply:
This would also do strange things with extensions that have changed type (e.g. enable an extension, which should not)